lib/geo_combine/ogp.rb

```diff
@@ -0,0 +1,131 @@
+module GeoCombine
+  # Data model for ESRI's open data portal metadata
+  class OGP
+    include GeoCombine::Formatting
+    attr_reader :metadata
+
+    ##
+    # Initializes an OGP object for parsing
+    # @param [String] metadata a valid serialized JSON string from an ESRI Open
+    # Data portal
+    def initialize(metadata)
+      @metadata = JSON.parse(metadata)
+    end
+
+    ##
+    # Creates and returns a Geoblacklight schema object from this metadata
+    # @return [GeoCombine::Geoblacklight]
+    def to_geoblacklight
+      GeoCombine::Geoblacklight.new(geoblacklight_terms.to_json)
+    end
+
+    ##
+    # Builds a Geoblacklight Schema type hash from Esri Open Data portal
+    # metadata
+    # @return [Hash]
+    def geoblacklight_terms
+      {
+        # Required fields
+        dc_identifier_s: metadata['LayerId'],
```

Comment:
Thanks for this. I've added URI encoding to both the ID and the slug. I'd like to see some more examples where we run into errors before we start adding more complexity here. Is that fair? Then we can add tests for those examples?